File: backend/src/main/java/com/simats/neoomfs/dto/response/ClinicalDecisionResponse.java
package com.simats.neoomfs.dto.response;

import com.simats.neoomfs.entity.ClinicalDecision;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

/**
 * Clinical decision response DTO.
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ClinicalDecisionResponse {
    private Long id;
    private Long patientId;
    private String patientName;
    private String patientMrn;
    private ClinicalDecision.RiskLevel riskLevel;
    private ClinicalDecision.FitnessDecision fitnessDecision;
    private Integer riskScore;
    private List<String> clinicalAlerts;
    private List<String> recommendations;
    private String decisionNotes;
    private String generatedByName;
    private LocalDateTime generatedAt;
}

File: frontend/app/src/main/java/com/simats/neoomfs/models/ClinicalDecisionResponse.kt
package com.simats.neoomfs.models

data class ClinicalDecisionResponse(
    val id: Long?,
    val patientId: Long?,
    val fitnessStatus: String? = null,
    val riskLevel: String? = null,
    val decisionNotes: String? = null,
    val redAlerts: List<String>? = null,
    val yellowAlerts: List<String>? = null,
    val recommendations: List<String>? = null
)
